### ISSUE TITLE
DLL load failed: The specified procedure could not be found.

### ISSUE THREAD

**Issue by stuft2** [byu-oit/awslogin#6]
I installed Python 3.6 from the Python Release page and then ran ```pip3 install byu-awslogin```
It says it installed correctly so I ran ```awslogin``` and I got the following error:
```
Traceback (most recent call last):
  File "c:\python36\lib\runpy.py", line 193, in _run_module_as_main
    "__main__", mod_spec)
  File "c:\python36\lib\runpy.py", line 85, in _run_code
    exec(code, run_globals)
  File "C:\Python36\Scripts\awslogin.exe\__main__.py", line 5, in <module>
  File "c:\python36\lib\site-packages\byu_awslogin\index.py", line 9, in <module>
    from .adfs_auth import authenticate
  File "c:\python36\lib\site-packages\byu_awslogin\adfs_auth.py", line 9, in <module>
    import lxml.etree as ET
ImportError: DLL load failed: The specified procedure could not be found.
```
My computer OS is Windows 10 64 bit
I also have python 2.7 and python 3.4 installed but currently I am using 3.6

I looked at https://stackoverflow.com/questions/16296050/error-using-etree-in-lxml and I've tried running 
```
python -m pip uninstall lxml
python -m pip install lxml==3.6.0
```
lxml uninstalled successfully but on the install I got this error:
```
  *********************************************************************************
  Could not find function xmlCheckVersion in library libxml2. Is libxml2 installed?
  *********************************************************************************
  error: command 'C:\\Program Files (x86)\\Microsoft Visual Studio 14.0\\VC\\BIN\\cl.exe' failed with exit status 2
```

**Comment by pauldeden** [comment]
@SpencerTuft would you run the following on your machine and post the output here?

`pip3 --version`

**Comment by ylehilds** [comment]
Just my 2 cents, would running the following command make any difference?

pip3.6 install byu-awslogin

**Comment by stuft2** [comment]
@pauldeden ```pip3 --version``` returns:
```
pip 9.0.1 from c:\python36\lib\site-packages (python 3.6)
```

**Comment by stuft2** [comment]
@ylehilds I get the same error as I did before.

**Comment by pauldeden** [comment]
Good to know. Thanks.

**Comment by pauldeden** [comment]
Installing lxml‑3.8.0‑cp36‑cp36m‑win32.whl from http://www.lfd.uci.edu/~gohlke/pythonlibs/#lxml and that worked.  Working on documenting the fix.

**Comment by pauldeden** [comment]
Documented the fix and saved the needed binary in https://github.com/byu-oit/awslogin/releases/tag/lxml

**Comment by stuft2** [comment]
Ran into another issue with awslogin. This time, I was helping someone else install awslogin. After installing awslogin and trying to run it, I got the error that says to go to https://github.com/byu-oit/awslogin/releases/tag/lxml to install lxml.  I got the following error when trying to install this version of lxml.

![image](https://user-images.githubusercontent.com/20521760/29290896-0069ceae-80ff-11e7-9147-ee5fcf311dba.png)

**Comment by pauldeden** [comment]
Reopening issue as there is a case just reported that wasn't resolved in the previous resolution.

**Comment by ndpete** [comment]
I've tried to recreate this a couple of times on fresh installs of windows and python and haven't been able to reproduce this. 

Any ideas on how I can reproduce this?
What version of python the 32 or 64 bit versions?

**Comment by ndpete** [comment]
can you post the output of `pip list` on the affected machine?

**Comment by ndpete** [comment]
Looks like the issue was caused by trying to install 32 bit wheel with 64 bit version of python. Upon trying to install the 64 bit wheel we also failed due to `(1)` in the file name. It appears when pip attempts to install a wheel it parses the filename to determine if it can install or not.
 
See this stack overflow question and answer for more information: [StackOverflow](https://stackoverflow.com/questions/28568070/filename-whl-is-not-supported-wheel-on-this-platform)
